src/main/java/com/microsoft/sqlserver/jdbc/IOBuffer.java

```diff
@@ -5337,10 +5338,15 @@ void writeRPCDateTime(String sName, GregorianCalendar cal, int subSecondNanos,
                 TDS.BASE_YEAR_1900);
 
         // Next, figure out the number of milliseconds since midnight of the current day.
-        int millisSinceMidnight = (subSecondNanos + Nanos.PER_MILLISECOND / 2) / Nanos.PER_MILLISECOND + // Millis into
-                                                                                                         // the current
-                                                                                                         // second
-                1000 * cal.get(Calendar.SECOND) + // Seconds into the current minute
+        int millisSinceMidnight;
+        // Millis into the current second
+        if (con.getServerMajorVersion() >= SQL_SERVER_VERSION_2016) {
```

Comment:
can you please put a little comment explaining why we have to do different things for different sql server versions?

Reply:
and can we do that just once in the beginning so we don't have to check every time?